src/Apps/Search/routes.tsx

```diff
@@ -7,30 +7,14 @@ import { SearchResultsCollectionsRouteFragmentContainer as SearchResultsCollecti
 import { SearchResultsGalleriesRouteRouteFragmentContainer as SearchResultsGalleriesRoute } from "Apps/Search/Routes/Galleries/SearchResultsGalleries"
 import { SearchResultsShowsRouteRouteFragmentContainer as SearchResultsShowsRoute } from "Apps/Search/Routes/Shows/SearchResultsShows"
 import { RouteConfig } from "found"
-import qs from "qs"
 import React from "react"
 import { graphql } from "react-relay"
 import { Provider } from "unstated"
 import { FilterState } from "./FilterState"
 import { SearchAppFragmentContainer as SearchApp } from "./SearchApp"
 
-// FIXME: The initial render includes `location` in props, but subsequent
-// renders (such as tabbing back to this route in your browser) will not.
 const prepareVariables = (params, props) => {
-  let paramsFromUrl = props.location ? props.location.query : {}
-  if (
-    Object.keys(paramsFromUrl).length === 0 &&
-    Object.keys(params).length === 0
-  ) {
-    paramsFromUrl = qs.parse(location.search.replace(/^\?/, ""))
```

Comment:
Locally, w/o continuing to do this whole section, I'm still seeing the errors with a missing term/keyword when doing tab navigation. Also, the only reason why initial page load is working b/c of the 

```
const paramsFromUrl = props.location ? props.location.query : {}
```

line below.

So basically...I'm still just seeing the exact same query param behavior (both linked and in storybooks).

Let's pair tomorrow? I'm not quite sure what this query param middleware is doing but I don't think it's quite hooked up right.